cypress/component/unit/ajax/User.spec.ts

```diff
@@ -0,0 +1,795 @@
+        } catch (err) {
+          expect(err.response.status).to.eq(404);
+        }
+      });
+    });
+  });
+
+  describe('getById', () => {
+    beforeEach(() => {
+      cy.intercept('GET', 'api/user/*', testUser1).as('getById');
+    });
+    it('uses an auth token in the request header', () => {
+      User.getById(testUser1.userId);
+      cy.wait('@getById').then((interception) => {
+        expect(interception.request.headers.authorization).to.eq(
+          'Bearer token'
+        );
+      });
+    });
+    it('sends a request to the correct endpoint', () => {
+      User.getById(testUser1.userId);
+      cy.wait('@getById').then(async (interception) => {
+        expect(interception.request.url).to.eq(
+          `${await getApiUrl()}/api/user/${testUser1.userId}`
+        );
+      });
+    });
+    describe('on success', () => {
+      it('returns the body of the response', async () => {
+        const returnedUser: DuosUserResponse = await User.getById(
+          testUser1.userId
+        );
+        expect(returnedUser.displayName).to.eq(testUser1.displayName);
+        expect(returnedUser.userId).to.eq(testUser1.userId);
+      });
+    });
+    describe('on rejection', () => {
+      it('does not handle the rejection', async () => {
+        cy.intercept('GET', 'api/user/*', { statusCode: 404 }).as('getById');
+        try {
+          await User.getById(testUser1.userId);
+        } catch (err) {
+          expect(err.response.status).to.eq(404);
+        }
+      });
+    });
+  });
+
+  describe('list', () => {
+    beforeEach(() => {
+      cy.intercept('GET', 'api/user/role/*', [testUser1, testUser2]).as('list');
+    });
+    it('uses an auth token in the request header', () => {
+      User.list('Admin');
+      cy.wait('@list').then((interception) => {
+        expect(interception.request.headers.authorization).to.eq(
+          'Bearer token'
+        );
+      });
+    });
+    it('sends a request to the correct endpoint', () => {
+      User.list('Admin');
+      cy.wait('@list').then(async (interception) => {
+        expect(interception.request.url).to.eq(
+          `${await getApiUrl()}/api/user/role/Admin`
+        );
+      });
+    });
+    describe('on success', () => {
+      it('returns the body of the response', async () => {
+        const returnedUsers: DuosUserResponse[] = await User.list('Admin');
+        expect(returnedUsers.length).to.eq(2);
+        expect(returnedUsers[0].displayName).to.eq(testUser1.displayName);
+        expect(returnedUsers[0].userId).to.eq(testUser1.userId);
+        expect(returnedUsers[1].displayName).to.eq(testUser2.displayName);
+        expect(returnedUsers[1].userId).to.eq(testUser2.userId);
+      });
+    });
+    describe('on rejection', () => {
+      it('does not handle the rejection', async () => {
+        cy.intercept('GET', 'api/user/role/*', { statusCode: 404 }).as('list');
+        try {
+          await User.list('Admin');
+        } catch (err) {
```

Comment:
Do we need anything before the catch to ensure that we actually did hit the assert in the catch block? This pattern does test as expected, so I'm fine with this as is.

Reply:
That's a good point, I'll put some `assert(false)`s after the ajax calls before the catch block to make sure it fails if it does not throw for some reason